build-logic/testing-convention/src/main/kotlin/convention.gradle-testing.gradle.kts

```diff
@@ -34,12 +34,40 @@ val gradleTestTask = tasks.register<Test>("gradleTest") {
 
     useJUnitPlatform()
 
-    @Suppress("MagicNumber")
-    maxParallelForks = 8
+    /**
+     * timings provided only as relative to each other
+     * gradleTest:
+     *  1: 2m14s
+     *  16: 3m16s
+     */
+    maxParallelForks = 1
+
+    /**
+     * usually there is a small amount of test classes per module, and tests are not so memory hungry
+     */
+    setForkEvery(null)
+
+    jvmArgs = listOf("-XX:+UseGCOverheadLimit", "-XX:GCTimeLimit=10")
+
+    minHeapSize = "128m"
+    maxHeapSize = "256m"
 
     failFast = false
 
+    /**
+     * fix for multiple `WARNING: Illegal reflective access`
```

Comment:
I'd like to clarify one point.

These parameters are also applied in convention plugin for unit tests:

```kotlin
tasks.withType<Test>().configureEach {
```

If I'm not mistaken, the unit tests convention plugin applies to all modules through the kotlin convention plugin.
Thus, it's not clear why we duplicate the same configuration in `tasks.register<Test>`.
I don't want such implicit overriding but I'd expect it from the code. 
If it's accidental, such relations between convention plugins is error-prone.
Please correct me if I'm wrong.

Reply:
You're absolutely right.
I removed all duplicates and found an error in setting jvmArgs, which is replacing extra(added in buildscript) set of args, instead of adding to it

The only problem left is values overwrite.
For example `maxParallelForks` set to `halfofavailableProcessors` in unit tests and then overwritten configuring `gradleTest` task with value `1`. It works that way only because of plugin apply ordering, which is error prone.

Reply:
> It works that way only because of plugin apply ordering, which is error prone

Can we check this contract explicitly inside gradle testing convention plugin? (`withPlugin` or similar constructs)